server/src/main/java/org/opensearch/telemetry/tracing/ThreadContextBasedTracerContextStorage.java

```diff
@@ -68,7 +68,7 @@ public Map<String, String> headers(Map<String, Object> source) {
 
         if (source.containsKey(CURRENT_SPAN)) {
             final SpanReference current = (SpanReference) source.get(CURRENT_SPAN);
-            if (current != null) {
+            if (current != null && current.getSpan() != null) {
```

Comment:
Please revert SpanReference should never refer to a `null` spans

Reply:
> Please revert SpanReference should never refer to a `null` spans

This is happening is certain test scenarios. Otel should always provide the propagated ROOT span.

Reply:
> This is happening is certain test scenarios. Otel should always provide the propagated ROOT span.

There should be not reasons to create a reference for `null` spans, there should be no SpanReference instead

Reply:
Yes, there is a scenario when we have the span and on detaching the previous span is null. The put code is just updating the SpanReference without checking the null value. Earlier this was handled this way - https://github.com/opensearch-project/OpenSearch/commit/dbb868a8f55878540c6b5a41f9c7a47f7644dcc0#diff-bf58bdcc0cc97e6e34dfc1736fd85d80628fb466eb9278ba6626b2f09e081d6bL43 but since now we need to update the span to null to clear up the context so let me change it and set null if span is null.

Reply:
We just need to remove the reference if there is no span in context anymore

Reply:
Ah! now I am able to recall, Updating the value to transienHeader doesn't work as there is a check ThreadContext$ThreadContextStruct.putSingleHeader and give error when we try to set null for the existing current_span value. Thats why had to update the existing SpanReference object.

Reply:
:( yeah, now me too, remember that :( I think no choice here but do `null` check :(